app/code/core/Mage/Catalog/controllers/ProductController.php

```diff
@@ -126,6 +126,8 @@ public function viewAction()
                 } elseif (!$this->getResponse()->isRedirect()) {
                     $this->_forward('noRoute');
                 }
+            } elseif (Mage::getIsDeveloperMode()) {
+                Mage::printException($e);
```

Comment:
Missing redirection here

Reply:
Yes, `printException` stop the process, so it's useless to redirect.